attachments/LVV-T2117.ipynb

```diff
@@ -0,0 +1,233 @@
+      "text/plain": [
+       "<Figure size 1440x720 with 2 Axes>"
+      ]
+     },
+     "metadata": {
+      "needs_background": "light"
+     },
+     "output_type": "display_data"
+    }
+   ],
+   "source": [
+    "def running_mean(x, N):\n",
+    "    cumsum = numpy.cumsum(numpy.insert(x, 0, 0)) \n",
+    "    return (cumsum[N:] - cumsum[:-N]) / float(N)\n",
+    "\n",
+    "tidx = numpy.where(numpy.array(diff)<20.)\n",
+    "tdiff = numpy.array(diff)[tidx]\n",
+    "tdates = numpy.array(dates)[tidx]\n",
+    "fig, (ax0, ax1) = plt.subplots(nrows=1, ncols=2, figsize=(20, 10))\n",
+    "_ = fig.suptitle('Measured latency -- lsst.sal.MTM1M3.logevent_heartbeat (summit)')\n",
+    "scat = ax0.scatter(tdates, tdiff, alpha=0.7, s=0.5)\n",
+    "xl0 = ax0.set_xlabel('Time UTC')\n",
+    "yl0 = ax0.set_ylabel('now() - private_sndStamp')\n",
+    "rm = ax0.plot(tdates[499:-500], running_mean(tdiff, 1000), c='black')\n",
+    "hist = ax1.hist(tdiff, cumulative=-1, bins=100, density=True)\n",
+    "yl1 = ax1.set_ylabel('1 - Cumulative distribution')\n",
+    "xl1 = ax1.set_xlabel('Measured latency (s)')\n",
+    "xr1 = ax1.set_xlim(-0.25, 3)\n",
+    "\n",
+    "for gap in gaps:\n",
+    "    ax0.axvspan(gap[0].datetime, gap[1].datetime, alpha=0.5, color='grey')\n",
+    "vl = ax1.axvline(numpy.percentile(tdiff, 90), c='black', ls='solid', label=f'90th = {numpy.percentile(tdiff, 90):.2f}')\n",
+    "v2 = ax1.axvline(numpy.percentile(tdiff, 95), c='black', ls='dashed', label=f'95th = {numpy.percentile(tdiff, 95):.2f}')\n",
+    "v3 = ax1.axvline(numpy.percentile(tdiff, 99), c='black', ls='dotted', label=f\"99th = {numpy.percentile(tdiff, 99, interpolation='lower'):.2f}\")\n",
+    "v4 = ax1.axvline(numpy.median(tdiff), c='black', ls='dashdot', label=f'median = {numpy.median(tdiff):.2f}')\n",
+    "v4 = ax1.axvline(6, c='black', ls='dashdot', alpha=0.0, label=f'max = {numpy.max(tdiff):.2f}')\n",
+    "leg = ax1.legend()\n",
+    "plt.gcf().autofmt_xdate()"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "c228b119",
+   "metadata": {},
+   "source": [
+    "The left pane shows the latency as a function of time.\n",
```

Comment:
Do we know why after the dropout does the mean and variance of the spread change? It's all within limits, but is there any way to know if it was just M1M3 behaving differently?

Reply:
The difference is that before the M1M3 subsystem was in the `DISABLED` state, but after that it's in the `ENABLED` state.  That is essentially measuring the effect of the M1M3 system publishing at a nominal, quiescent rate.  We couldn't get a 5 day run where the M1M3 was completely off, but I think this answers the spirit of the test case that we look at the behavior of the low latency topics in a quiescent phase if not completely quiet.

Reply:
Sounds like you do know the reason so I'm satisfied with this.